llarp/router/outbound_message_handler.cpp

```diff
@@ -0,0 +1,209 @@
+#include <router/outbound_message_handler.hpp>
+
+#include <messages/link_message.hpp>
+#include <router/i_outbound_session_maker.hpp>
+
+namespace llarp
+{
+
+  bool
+  OutboundMessageHandler::QueueMessage(const RouterID &remote,
+                                       const ILinkMessage *msg,
+                                       SendStatusHandler callback)
+  {
+    std::array< byte_t, MAX_LINK_MSG_SIZE > linkmsg_buffer;
+    llarp_buffer_t buf(linkmsg_buffer);
+
+    if (!EncodeBuffer(msg, buf))
+    {
+      return false;
+    }
+
+    Message message = std::make_pair(buf.sz, callback);
+
+    memcpy(message.first.data(), buf.base, buf.sz);
```

Comment:
`std::copy_n` instead?

Reply:
Yeah, I wasn't 100% sure about this bit.  Feels weird to copy into a vector for some reason...but this is how it was being done in router.cpp so I kept it this way.

Reply:
changed.